lib/index.js

```diff
@@ -970,7 +1000,7 @@ GA.prototype.productListFilteredEnhanced = function(track) {
  * @param {Track} track
  */
 
-function enhancedEcommerceTrackProduct(track) {
+function enhancedEcommerceTrackProduct(track, context) {
```

Comment:
dont use the keyword `context` for variable names unless you are actually referring to the `context` object of the segment raw payload since that's just the convention.

instead of passing the entire `this` as a param, just pass either the value you need or look up how to use `bind`